[sqlite] Initial plan and tests

MindNumbing opened the pull request:
Signed-off-by: Steven Marshall <SMarshall@chef.io>

Commit:
[sqlite] Initial plan and tests

Signed-off-by: Steven Marshall <SMarshall@chef.io>

gavindidrichsen commented on controls/sqlite_works.rb:
```diff
@@ -0,0 +1,37 @@
+    its('exit_status') { should eq 0 }
+  end
+
+  sqlite_works = command("/bin/sqlite3 --version")
```
Can you make this the full /hab/pkgs/ORIGIN/sqlite/VERSION/TIMESTAMP/bin/sqlite3 path please.  The full path will verify that the newly built binary is actually being tested.

Commit:
[sqlite] Various refactors of existing code

Added inspec.lock to .gitignore
Updated README.md to latest binary format
Added plan_name to attributes.yml
Updates controls to add more prescriptive version checking.
Updated inspec.yml to latest version

Signed-off-by: Steven Marshall <SMarshall@chef.io>

gavindidrichsen reviewed (approved):
LGTM
